students/ethan_nguyen/lesson04/trigram.py

```diff
@@ -0,0 +1,97 @@
+
+import sys
+import random, os
+
+
+def build_trigrams(words):
+    '''
+    build up the trigrams dict from the list of words
+    returns a dict with:
+       keys: word pairs
+       values: list of followers
+    '''
+    
+    trigrams = {}
+
+    for i in range(len(words)-2): 
+        pair = tuple(words[i:i + 2]) #make trigrams key as tuple
+        follower = words[i + 2] #make value as list
+
+        trigrams.setdefault(pair,[]).append(follower) #create key with value if key not already exist
+
+    return trigrams
+
+def make_words(words):
+    
+    return words.split(" ")
+
+def build_text(word_pairs):
+
+    '''
+    build up a new story with the trigram dict
+    '''
+
+    story_words = list()
+    #randomely select two staring words
+    start_key = random.choice(list(word_pairs))
+    [story_words.append(item) for item in start_key]
```

Comment:
this should actually be:
```

story_words = [item for item in start_key]
```
a comprehension should include an assignment, if there is no assignment then it should be a regular loop.

In above case you don't really need a comprehension, `start_key` is a tuple and looks like you want to initialize a list with that? this is what you want
```
story_words.extend(list(start_key))
```